app.py

```diff
@@ -230,6 +251,39 @@ def delete_user():
     return redirect("/signup")
 
 
+@app.route('/users/add_like/<int:message_id>', methods=['POST'])
+def toggle_like(message_id):
+    """Toggle a liked message for the current user."""
+
+    if not g.user:
+        flash("Access unauthorized.", "danger")
+        return redirect("/")
+
+    liked_message = Message.query.get(message_id)
+    if liked_message.user_id == g.user.id:
+        return abort(403)
```

Comment:
I think you meant to do this?
```suggestion
    if liked_message.user_id != g.user.id:
        return abort(403)
```